symfony/plugins/orangehrmTimePlugin/Dao/ProjectActivityDao.php

```diff
@@ -0,0 +1,138 @@
+    ): Paginator {
+        $q = $this->createQueryBuilder(ProjectActivity::class, 'projectActivity');
+        $q->andWhere('projectActivity.project = :projectId');
+        $q->andWhere('projectActivity.deleted = :status');
+        $q->setParameter('projectId', $projectId);
+        $q->setParameter('status', false);
+        $this->setSortingAndPaginationParams($q, $projectActivitySearchFilterParams);
+        return $this->getPaginator($q);
+    }
+
+    /**
+     * @param int $projectId
+     * @param ProjectActivitySearchFilterParams $projectActivitySearchFilterParams
+     * @return int
+     */
+    public function getProjectActivityCount(
+        int $projectId,
+        ProjectActivitySearchFilterParams $projectActivitySearchFilterParams
+    ): int {
+        $paginator = $this->getProjectActivitiesByProjectId($projectId, $projectActivitySearchFilterParams);
+        return $paginator->count();
+    }
+
+    /**
+     * @param ProjectActivity $projectActivity
+     * @return ProjectActivity
+     */
+    public function saveProjectActivity(ProjectActivity $projectActivity): ProjectActivity
+    {
+        $this->persist($projectActivity);
+        return $projectActivity;
+    }
+
+    /**
+     * @param int $projectId
+     * @return Project|null
+     */
+    public function getProjectById(int $projectId): ?Project
+    {
+        $project = $this->getRepository(Project::class)->find($projectId);
+        return ($project instanceof Project) ? $project : null;
+    }
+
+    /**
+     * @param int $projectId
+     * @param int $projectActivityId
+     * @return ProjectActivity|null
+     */
+    public function getProjectActivityByProjectIdAndProjectActivityId(
+        int $projectId,
+        int $projectActivityId
+    ): ?ProjectActivity {
+        $project = $this->getProjectById($projectId);
+        $projectActivity = $this->getRepository(ProjectActivity::class)->findOneBy(
+            ['activityId' => $projectActivityId, 'project' => $project]
+        );
+        return ($projectActivity instanceof ProjectActivity) ? $projectActivity : null;
+    }
+
+    /**
+     * @param int $activityId
+     * @return bool
+     */
+    public function hasActivityGotTimesheetItems(int $activityId): bool
+    {
+        $q = $this->createQueryBuilder(TimeSheetItem::class, 'timeSheetItem');
+        $q->andWhere('timeSheetItem.projectActivity = :projectId');
+        $q->setParameter('projectId', $activityId);
+        $count = $this->getPaginator($q)->count();
+        return ($count > 0);
+    }
+
+    /**
+     * @param int[] $toBeDeletedActivityIds
+     * @throws ProjectServiceException
+     */
+    public function deleteProjectActivities(array $toBeDeletedActivityIds)
+    {
+        $delete = true;
+        foreach ($toBeDeletedActivityIds as $toBeDeletedActivityId) {
+            $deletable = $this->hasActivityGotTimesheetItems($toBeDeletedActivityId);
```

Comment:
you can simplify this 
```
if(! $this->hasActivityGotTimesheetItems($toBeDeletedActivityId)) {
  throw ProjectServiceException::projectActivityExist();
}
```

Reply:
foreach ($toBeDeletedActivityIds as $toBeDeletedActivityId) {
            if ($this->hasActivityGotTimesheetItems($toBeDeletedActivityId)) {
                throw ProjectServiceException::projectActivityExist();
            }
        }

Reply:
![image](https://user-images.githubusercontent.com/89253437/144223552-e09c8dbb-322c-416a-a9ee-7b5f4cfc1e99.png)